### DISCUSSION TITLE
Can this be translated?

### DISCUSSION THREAD

**Discussion by aljawaid** [aljawaid/KanboardEmailHistory#10] [Q&A] [answered]
`'send_to' => array(... `

**Comment by aljawaid** [comment, marked as answer]
@creecros 

> `'send_to' => array(... `

is being picked up in `params.php` as
```
<?= $this->form->label(ucfirst($param_name), $param_name) ?>
```

which makes it `Send_to` .... can it be translated in the code, or should I remove the underscore via php in the `params.php` ?

**Reply by creecros** [reply]
I dont know what you are trying to do.

**Reply by aljawaid** [reply]
I want to translate `send_to` as the underscore shows in the frontend

**Reply by creecros** [reply]
ya still not following. where does it show up?

**Reply by aljawaid** [reply]
In the actions page once the action is set in the project. Will send a screenshot when I get home

**Reply by aljawaid** [reply]
![image](https://user-images.githubusercontent.com/10233708/212491980-128cdfb7-e0e1-46d2-87fb-d7adcee9e300.png)

**Reply by creecros** [reply]
got it. Um, ya. Since you are using your own param.php, just add an if then else. 

again on my cell phone, so I'll write it out basic.

if $param_name = send_to
$this->form->label(t('Send to'), $param_name) 
else
$this->form->label(ucfirst($param_name), $param_name)

**Reply by aljawaid** [reply]
> got it. Um, ya. Since you are using your own param.php, just add an if then else.
> 
> again on my cell phone, so I'll write it out basic.
> 
> if $param_name = send_to $this->form->label(t('Send to'), $param_name) else $this->form->label(ucfirst($param_name), $param_name)

Does this look right?
```php
<?php elseif (is_array($param_desc)): ?>
            <?php if ($param_name = 'send_to'): ?>
                <?= $this->form->label(t('Send to'), $param_name) ?>
            <?php else: ?>
                <?= $this->form->label(ucfirst($param_name), $param_name) ?>
        <?= php endif ?>
            <?= $this->form->select('params['.$param_name.']', $param_desc, $values) ?>
        <?php else: ?>
```

**Reply by aljawaid** [reply]
I have done this but it is useless until I can figure out how to 'hook' it or work with 'autosubtasks' (which I also need) #12

**Reply by creecros** [reply]
Here is an example of a hook:

https://github.com/kanboard/kanboard/blob/5750e273bc2154ce04ac8cbb804d0b1ca3692620/app/Template/task/show.php#L1

when you use the hook `template:task:show:top` that is where it inserts the code. 

now notice, there are no hooks in params.php

https://github.com/kanboard/kanboard/blob/main/app/Template/action_creation/params.php

so when I say, you need to add a hook or hooks, this is where you need a hook. somewhere in this code. 

or, forget the hook, and just add the if statement to AutoSubtasks in the Plugin.php to not use any template override of params.php, in the event of your plugins existence.

**Comment by aljawaid** [comment]
> or, forget the hook, and just add the if statement to AutoSubtasks in the Plugin.php to not use any template override of params.php, in the event of your plugins existence.
@creecros 
I could do this, from a fork of Autosubtasks I guess - it might be the better way, but please check this comment, I think this might be better if it works as it keeps all the work in thsi plugin and doesnt affect others... https://github.com/aljawaid/KanboardEmailHistory/discussions/12#discussioncomment-4689191
